pkg/apis/pipeline/v1beta1/customrun_validation.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/validate"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"knative.dev/pkg/apis"
+)
+
+var _ apis.Validatable = (*CustomRun)(nil)
+
+// Validate taskrun
+func (r *CustomRun) Validate(ctx context.Context) *apis.FieldError {
+	if err := validate.ObjectMetadata(r.GetObjectMeta()).ViaField("metadata"); err != nil {
+		return err
+	}
+	if apis.IsInDelete(ctx) {
+		return nil
+	}
+	return r.Spec.Validate(ctx)
+}
+
+// Validate CustomRun spec
+func (rs *CustomRunSpec) Validate(ctx context.Context) *apis.FieldError {
+	// this covers the case rs.Ref == nil && rs.Spec == nil
+	if equality.Semantic.DeepEqual(rs, &CustomRunSpec{}) {
+		return apis.ErrMissingField("spec")
+	}
+
+	if rs.CustomRef != nil && rs.CustomSpec != nil {
+		return apis.ErrMultipleOneOf("spec.ref", "spec.spec")
```

Comment:
spec.CustomRef && spec.CustomSpec? same for the rest of this func.

Reply:
done.